core/templates/pages/contributor-dashboard-page/contributor-dashboard-page.component.ts

```diff
@@ -38,21 +38,26 @@ import { WindowRef } from 'services/contextual/window-ref.service';
 })
 export class ContributorDashboardPageComponent
   implements OnInit {
-  OPPIA_AVATAR_LINK_URL: string = AppConstants.OPPIA_AVATAR_LINK_URL;
-  defaultHeaderVisible: boolean;
-  profilePictureDataUrl: SafeUrl | string;
-  username: string;
-  userInfoIsLoading: boolean;
-  userIsLoggedIn: boolean;
-  userIsReviewer: boolean;
-  userCanReviewTranslationSuggestionsInLanguages: string[];
-  userCanReviewVoiceoverSuggestionsInLanguages: string[];
-  userCanReviewQuestions: boolean;
-  tabsDetails: ContributorDashboardTabsDetails;
-  OPPIA_AVATAR_IMAGE_URL: string;
-  languageCode: string;
-  topicName: string;
-  activeTabName: string;
+  OPPIA_AVATAR_LINK_URL: string | null = AppConstants.OPPIA_AVATAR_LINK_URL;
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion. For more information, see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  defaultHeaderVisible!: boolean;
+  profilePictureDataUrl!: SafeUrl | string;
+  // The following is property is set to null when the
+  // user is not logged in.
+  username!: string | null;
```

Comment:
If this can be `null`, shouldn't we remove the non-null (!) assertion operator? Ditto elsewhere.

Reply:
Done